woocommerce/woocommerce
# Add RichTextEditor component using native block editor toolbar

### All Submissions:






### Changes proposed in this Pull Request:

This PR is an alternative to https://github.com/woocommerce/woocommerce/pull/34629.

Instead of creating and maintaining our own toolbar, we use a fixed block editor toolbar and prevent deselection of blocks so that it's always shown.

This needs to be tested in conjunction with https://github.com/WordPress/gutenberg/issues/44508

Note that there are still 2 issues that will need to be addressed in a follow-up:

1. There is some odd behavior around changing of blocks where sometimes it needs to be done a couple times in order to actually change an element.
2. The fix that initially creates an empty paragraph so that we can keep the toolbar visible also focuses the editor immediately.  This could lead to unexpected results and poor accessibility.

### How to test the changes in this Pull Request:

1. Check out and activate this branch of GB - https://github.com/WordPress/gutenberg/pull/44517
2. Import and use this editor in WCA somewhere (e.g., `/client/homescreen/layout.js`)
```
import { RichTextEditor } from '@woocommerce/components';

<RichTextEditor onChange={ () => console.log('changed') } />
```
3. Enqueue the styles for the editor:
```
		wp_enqueue_style( 'wp-edit-post' );
		wp_enqueue_style( 'wp-format-library' );
		wp_enqueue_editor();
		do_action( 'enqueue_block_editor_assets' );
```
4. Play around with the editor and make sure it functions as expected

### Other information:




### FOR PR REVIEWER ONLY:
